virtual_ecosystem/models/plants/plants_model.py

```diff
@@ -164,9 +173,25 @@ def __init__(
         self.model_constant: PlantsConsts
         """Set of constants for the plants model"""
         self.communities: PlantCommunities
-        """Initialise the plant communities from the data object."""
-        self._canopy_layer_indices: np.ndarray
-        """The indices of the canopy layers within wider vertical profile"""
+        """An instance of PlantCommunities providing dictionary access keyed by cell id
+        to PlantCommunity instances for each cell."""
+        self._canopy_layer_indices: NDArray[np.bool_]
+        """The indices of the canopy layers within wider vertical profile. This is 
+        a shorter reference to self.layer_structure.index_canopy."""
+        self.canopies: dict[int, Canopy]
+        """A dictionary giving the canopy structure of each grid cell."""
+        self.filled_canopy_mask: NDArray[np.bool_]
+        """A boolean array showing which layers contain canopy by cell."""
+        self.stem_gpp: dict[int, NDArray[np.float32]]
+        """A dictionary keyed by cell id giving the stem GPP for each cohort in the cell
+        community"""
+        self.stem_transpiration: dict[int, NDArray[np.float32]]
+        """A dictionary keyed by cell id giving the stem transpiration for each cohort
+        in the cell community"""
+        self.pmodel_consts: PModelConst
+        """PModel constants used by pyrealm."""
+        self.pmodel_core_consts: CoreConst
+        """Core constants used by pyrealm."""
```

Comment:
is there much overlap with the ve core constants and if yes, should priority be given to ve? Something to think about at some later point

Reply:
Yes - there likely is quite a bit of overlap.